### PR TITLE
Remove secondary/unused dependencies

### PR DESCRIPTION
**Pull request by pmav99**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,32 +7,25 @@ readme = "README.md"
...
-eccodes = "*"
-fastparquet = "*"
-fsspec = "*"
+dask = {version = "*", extras = ["complete"]}
 geopandas = "*"
-jupyter-bokeh = "*"
-lttb = "*"
 matplotlib = "*"
 observer = {git = "https://github.com/seareport/observer.git"}
 owslib = "*"
```
Comment (tomsail) [inline pyproject.toml:15]: you can remove owslib

File: pyproject.toml [hunk 2]
```diff
@@ -7,32 +7,25 @@ readme = "README.md"
...
 pandas = "*"
-pyarrow = "*"
 pyposeidon = {git = "https://github.com/ec-jrc/pyPoseidon.git"}
 ruptures = "*"
-scikit-image = "*"
```
Comment (tomsail) [inline pyproject.toml:26]: this require to also remove https://github.com/seareport/analysea/blob/f3f863a9424ea7ae17d75f50443c50fbcbe1044c/analysea/steps.py#L12

**Comment by pmav99** [comment]
Is there some reason we install thalassa from git and not from pypi?

**Comment by pmav99** [comment]
I think that the pre-commit failure is not related to the changes. Something network related. It should get resolved in time. 

Some rationale behind this PR: 

`azure-identity`, `adlfs`, `fsspec`, `pyarrow`, `fastparquet` etc should be pulled by `observer`. Since we don't explicitly import them here, there is no need to have them as primary dependencies. The idea is that if `observer` decides to switch some library or something, the change only happens on observer and not here (assuming the API remains the same of course).

The exception is `xarray` and `dask`. Unfortunately, `poetry` does not pull nested "extras" dependencies. This means that even though `xarray[io,accel]` should pull `dask[complete]`, this does not happen. If we want to have `dask[complete]` we need to do it explicitly ourselves. That's why I added both in `pyproject.toml`.

**Review by tomsail — APPROVED** [review]

**Pull request merged**